server/asset/eth/common.go

```diff
@@ -31,6 +37,19 @@ const (
 	GweiFactor = 1e9
 )
 
+// TxIDFlag and SwapFlag are used in CoinIDs to signify a coinID
+// as either a transaction ID or a combination of a swap contract
+// address and secret hash. One or the other must be set.
+const (
+	// TxID indicates that this coin ID's hash is a txid. The address
+	// portion is zeros and unused.
+	TxIDFlag CoinIDFlag = 1 << iota
+	// SwapFlag indicates that this coin ID represents a swap with a
+	// contract address and secret hash used to fetch data about a swap
+	// from the live contract.
+	SwapFlag
```

Comment:
With enum value names I really like having a common prefix that assists with discovering usable values (e.g. IDE completion).  e.g. `SigHashAll`, `SigHashNone`, `SigHashSingle`, etc. being various `SigHashType`s. e.g. `STEcdsaSecp256k1`, `STEd25519`, and `STSchnorrSecp256k1` being the various `SignatureType`s. e.g. all the `txscript.ScriptFlags`. Also same convention that linters like to suggest for error type names.

As in the dcrd examples above, the prefix can deviate from the type names for brevity.  Ideas

`CoinFlagTxID`
`CoinFlagSwap`

`CIDFlagTxID`
`CIDFlagSwap`

`CIDTxID`
`CIDFlagSwap`

Feelings about the general approach of prefixing or specific prefixes?

Reply:
Sounds Good. Will go with one of those.

Reply:
Will also change the SwapStates naming I guess.